pkg/dbutil/common.go

```diff
@@ -320,6 +371,138 @@ func GetCRC32Checksum(ctx context.Context, db *sql.DB, schemaName, tableName str
 	return checksum.Int64, nil
 }
 
+// Bucket saves the bucket information from TiDB.
+type Bucket struct {
+	Count      int64
+	LowerBound string
+	UpperBound string
+}
+
+// GetBucketsInfo SHOW STATS_BUCKETS in TiDB.
+func GetBucketsInfo(ctx context.Context, db *sql.DB, schema, table string, tableInfo *model.TableInfo) (map[string][]Bucket, error) {
+	/*
+		example in tidb:
+		mysql> SHOW STATS_BUCKETS WHERE db_name= "test" AND table_name="testa";
+		+---------+------------+----------------+-------------+----------+-----------+-------+---------+---------------------+---------------------+
+		| Db_name | Table_name | Partition_name | Column_name | Is_index | Bucket_id | Count | Repeats | Lower_Bound         | Upper_Bound         |
+		+---------+------------+----------------+-------------+----------+-----------+-------+---------+---------------------+---------------------+
+		| test    | testa      |                | PRIMARY     |        1 |         0 |    64 |       1 | 1846693550524203008 | 1846838686059069440 |
+		| test    | testa      |                | PRIMARY     |        1 |         1 |   128 |       1 | 1846840885082324992 | 1847056389361369088 |
+		+---------+------------+----------------+-------------+----------+-----------+-------+---------+---------------------+---------------------+
+	*/
+	buckets := make(map[string][]Bucket)
+	query := "SHOW STATS_BUCKETS WHERE db_name= ? AND table_name= ?;"
+	log.Debugf("GetBucketsInfo query: %s", query)
+
+	rows, err := db.QueryContext(ctx, query, schema, table)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	defer rows.Close()
+
+	cols, err := rows.Columns()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	for rows.Next() {
+		var dbName, tableName, partitionName, columnName, lowerBound, upperBound sql.NullString
+		var isIndex, bucketID, count, repeats sql.NullInt64
+
+		// add partiton_name in new version
+		switch len(cols) {
```

Comment:
can get these info from `mysql`.`stats_buckets`? if so, use `SELECT NEEDED COLUMNS FROM` mya have better compatibility.

Reply:
ok, I will check it

Reply:
the result of `SHOW STATS_BUCKETS` is transform from mysql. stats_buckets,  mysql. stats_buckets only have tableid, and don't have column name. it is difficult for me to do transform here, I think use `SHOW STATS_BUCKETS`   is more convenient.